semantic_release/cli.py

```diff
@@ -220,13 +220,21 @@ def should_bump_version(
     """Test whether the version should be bumped."""
     if new_version == match_version and not retry:
         logger.info("No release will be made.")
+        if os.getenv("GITHUB_ACTIONS"):
```

Comment:
This is repeated a few times - I personally disagree with "hidden" parameters (e.g. loading env vars) in a function, but I would consider writing a little helper like 
```python
def write_github_output(output: io.StringIO, released: bool, version: str) -> None:
    print(f'released={str(released).lower()}', file=output)
    print(f'version={match_version}', file=output)
```
You can also use a little trick (thanks, Python!) to condense the output logic if you want:

```python
with open(os.getenv("GITHUB_OUTPUT", os.devnull), "a") as f:
    ...
```
That will just send the output to the null file if "GITHUB_OUTPUT" isn't set; it strictly shouldn't be necessary to even check "GITHUB_ACTIONS".

The advantages are that you can open the file once and write to it multiple times, if need be, and that when testing you can directly pass a `io.StringIO` instance and check the content after calling this function - no files needed.

That said, I don't mind terribly if you'd prefer to just pass the filename as a string and open the file within the function - both should be simple enough to test (see comment on tests)